### ISSUE TITLE
Training FCOS and RetinaNet using multiple GPUs , each loss = nan

### ISSUE THREAD

**Issue by Tahy1** [open-mmlab/mmdetection#4292]
Hello, I was training a model using FCOS or RetinaNet, backbone was ResNet18 using multiple GPUs, however, its loss was "nan" after 50 to 300 iters. Sometimes loss can converge normally(1 /10)
Faster RCNN would not make that mistake. Then, I tried to train FCOS using single GPU, its loss was normal(lower than 2 after several iters). My learning rate is 0.01, 8 GPUs, 2 samples per GPU, warmup_iters=700. My GPUs are Tesla K80
Also, I tried to print losses druing training, including loss_cls, loss_bbox and loss_centerness, sometimes they change to "nan" suddenly, sometimes loss_cls change to "inf" or more than 1e4.
I doubt that it caused by Focal loss, then I print input X before sigmoid larger than -1. Unfortunately, only a few X larger than -1 before loss == "nan"
II can only train the model with a single GPU. I don't know what causes this problem. Could you please give me a help?

**Comment by yhcao6** [comment]
When you changing the number of GPUs, you also need to change the learning rate.
To avoid `NaN` at the beginning, you can try to add [`grad_clip`](https://github.com/open-mmlab/mmdetection/blob/43b1909221179395ebd36edd24540940274ab9c9/configs/fcos/fcos_r50_caffe_fpn_4x4_1x_coco.py#L97-L98) or decrease the [warm-up ratio](https://github.com/open-mmlab/mmdetection/blob/43b1909221179395ebd36edd24540940274ab9c9/configs/fcos/fcos_r50_caffe_fpn_4x4_1x_coco.py#L100-L105).

**Comment by Tahy1** [comment]
> When you changing the number of GPUs, you also need to change the learning rate.
> To avoid `NaN` at the beginning, you can try to add [`grad_clip`](https://github.com/open-mmlab/mmdetection/blob/43b1909221179395ebd36edd24540940274ab9c9/configs/fcos/fcos_r50_caffe_fpn_4x4_1x_coco.py#L97-L98) or decrease the [warm-up ratio](https://github.com/open-mmlab/mmdetection/blob/43b1909221179395ebd36edd24540940274ab9c9/configs/fcos/fcos_r50_caffe_fpn_4x4_1x_coco.py#L100-L105).
Thanks for your reply. FCOS offical learning rate is 0.01 during 8GPUs and 2 imgs per GPU, it means that batch_size=16, I have the same lr as they do.
Last night I tried to set loss_cls, loss_bbox and loss_centerness equal to 0, separately. And "nan" issue wasn't happen only if loss_bbox=0. Then I print pos_bbox_preds, it shows that pred bboxes become a very large area beyond input image. So what make this happen? I will try grad_clip.

**Comment by Tahy1** [comment]
I tried to use grad_clip to eliminate gradiant and decrease the warm-up ratio in MULTIPLE-GPUS mode, but it didn't work, loss still became 'NaN' after several iters. However, training in SINGLE-GPU mode would not encounter this issue. I doublt MULTIPLE-GPUS training mode has some bugs while training FCOS, RetinaNet in specific dataset.

**Comment by huangruofei** [comment]
I have same issue,in single-GPU bbox_loss is nan.I change bbox_loss to CIOU_Loss. it is ok.

**Comment by ZwwWayne** [comment]
Please check available solutions in [FAQ about training](https://mmdetection.readthedocs.io/en/latest/faq.html#training).
